### PR TITLE
Event Interactions

### PR DESCRIPTION
**Pull request by jeda234**
Backend code related to liking/un-liking an event, sliding/un-sliding an event, and creating/editing/deleting a comment.
Changes included:
- Creating functions in users.services to update array containing liked and slid events for a unique user
- Creating functions in events.services to;
  - Update the event like and slide counter
  - Create a new comment
    - Update said comment
    - Delete said comment
- Creating a comment schema

### CODE REVIEW

File: backend/src/events/events.service.ts
Comment (dyamijr) [inline backend/src/events/events.service.ts:1]: All functions should return the eventid after a successful action. Comment functions should return the comment id.

File: backend/src/events/events.controller.ts [hunk 1]
```diff
@@ -10,14 +10,65 @@ import {
...
+  @Post(':id/unslide')
+  async unslideEvent(@Param('id') id: string, @Request() req) {
+    const event = await this.eventsService.unslideEvent(id, req.user);   
+    return event;
+  }
+
+  @UseGuards(AuthenticatedGuard)
+  @Post(':id/comment/new')
+  async commentEvent(@Param('id') id: string, @Body() comment: CommentDto, @Request() req) {
+    const event = await this.eventsService.commentEvent(id, comment.content, req.user);
```
Comment (dyamijr) [inline backend/src/events/events.controller.ts:51]: for the comment functions change event to comment just to clear up any possible confusion.

File: backend/src/schemas/user.schema.ts
Comment (dyamijr) [inline backend/src/schemas/user.schema.ts:1]: The likedEvents and slidEvents should be types of mongoose.Types.ObjectId[]. See comments or collaborators in event schema.

File: backend/src/events/events.service.ts [hunk 2]
```diff
@@ -20,18 +25,160 @@ export class EventsService {
...
+      );
+    }
+    if (await this.userService.findLike(id, uid)!=-1) { 
+      throw new UnauthorizedException(
+        'Already Liked',
+      );
+    } else {
+      event.likes++;
+      await event.save();
+      this.userService.likeEvent(id, uid);
```
Comment (dyamijr) [inline backend/src/events/events.service.ts:56]: for like event to take in a monoose object you should send the event._id

File: backend/src/users/users.service.ts [hunk 3]
```diff
@@ -34,4 +34,40 @@ export class UsersService {
     await createdUser.save();
     return createdUser;
   }
+
+  async likeEvent(id: String, uid: mongoose.Types.ObjectId) {
```
Comment (dyamijr) [inline backend/src/users/users.service.ts:38]: change id name to eventId and its type to mongoose.Types.ObjectId

File: backend/src/users/users.service.ts [hunk 4]
```diff
@@ -34,4 +34,40 @@ export class UsersService {
...
+    const user = await this.userModel.findById(uid);
```
Comment (dyamijr) [inline backend/src/users/users.service.ts:39]: should also add a check to make sure the user exist.

File: backend/src/users/users.service.ts [hunk 5]
```diff
@@ -34,4 +34,40 @@ export class UsersService {
...
+  }
+
+  async unslideEvent(id: String, uid: mongoose.Types.ObjectId) {
+    const user = await this.userModel.findById(uid);
+    user.slidEvents.splice(user.slidEvents.indexOf(id), 1);
+    await user.save();
+  }
+
+  async findSlide(id: String, uid: mongoose.Types.ObjectId) {
+    console.log('test')
```
Comment (dyamijr) [inline backend/src/users/users.service.ts:68]: remove this

File: backend/src/users/users.service.ts
Comment (dyamijr) [inline backend/src/users/users.service.ts:1]: apply changes mentioned in like event to all functions it applies to

**Review by dyamijr — CHANGES_REQUESTED** [review]
Nice job overall. Just fix the suggested changes and then this is good to go.

**Comment by jeda234** [comment]
Changes added

**Review by dyamijr — APPROVED** [review]
Looks great

**Pull request merged**